homeassistant/components/withings/__init__.py

```diff
@@ -179,8 +180,8 @@ async def register_webhook(
             webhook_url = await _async_cloudhook_generate_url(hass, entry)
         else:
             webhook_url = webhook_generate_url(hass, entry.data[CONF_WEBHOOK_ID])
-
-        if not webhook_url.startswith("https://"):
+        url = URL(webhook_url)
+        if url.scheme != "https" or url.port != 443:
             LOGGER.warning(
                 "Webhook not registered - "
                 "https and port 443 is required to register the webhook"
```

Comment:
Is 443 actually required or just https?

Reply:
I copied this initially from Netatmo, but I got an issue (see linked issue) where someone was hosting on a separate port and withings wouldn't want to create webhooks. So yes, 443 is required

Reply:
Thats a strange requirement to have. 🤷